plugins/GSdx/GSWndWGL.cpp

```diff
@@ -323,16 +331,45 @@ void* GSWndWGL::GetProcAddress(const char* name, bool opt)
 
 //TODO: check extensions supported or not
 //FIXME : extension allocation
-void GSWndWGL::SetVSync(int vsync)
+void GSWndWGL::SetSwapInterval(int vsync)
 {
 	// m_swapinterval uses an integer as parameter
 	// 0 -> disable vsync
 	// n -> wait n frame
 	if (m_swapinterval) m_swapinterval(vsync);
 }
 
+bool GSWndWGL::CompositorEnabled()
+{
+	BOOL dwm;
+	if (DwmIsCompositionEnabled(&dwm) != S_OK)
+		return false;
+
+	// Note: DWM is always true on Win8 and later
+	if (!dwm)
+		return false;
+
+	// Easy case are done. It remains driver black magic
+
+	// Recent driver (+380.xx ??) seem to always get v-sync enforced someway.
+	// XXX it would be nice to understand what happen to fix it properly
+	if (GLLoader::nvidia_buggy_driver)
+		return true;
+
+	// If the surface is fullscreen, the driver will trigger an exclusive
+	// fullscreen mode.
+	GSVector4i surface = GetClientRect();
+	GSVector4i window = GetWindowRect();
+	return !(surface == window).alltrue();
+}
+
 void GSWndWGL::Flip()
 {
+	// The compositor manages its own kind of Vsync/double-buffering. You need to wait the availabitiliy of the buffer
+	// before the swap to avoid stuttering.
+	if (CompositorEnabled() && vsync)
```

Comment:
mhh no, like - I meant seriously. 
Putting aside I'm skeptic about it, until two days ago I remember that read m_vsync or something.